app/models/ability.rb

```diff
@@ -0,0 +1,19 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    can :read, User, user: user
+    can :read, Post, user: user
+    can :read, Comment, user: user
+    can :read, Like, user: user
+    return unless user.present?
+
+    can %i[update destroy], Post, author_id: user.id
+    can %i[update destroy], Comment, author_id: user.id
+
+    return unless user.is? :admin
+
+    can :manage, :all
+  end
```

Comment:
- [ ] The privileges of an admin user is not properly establized in your authorization model. Kindly not that a  universal rule (validation) also covers sub rules and therefore both validations should not be set at the same time. Kindly improve upon this aspect by updating your Ability model similar to as below:

    ```ruby
     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
```